modules/stanford_profile_styles/lib/js/node_form.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @file
+ * Stops page from changing when user is posting.
+ */
+
+(function ($, Drupal) {
+  'use strict';
+  // Allow Save Submit button.
+  var save = false;
+  // Flag the form has been edited.
+  var edit = false;
+
+  Drupal.behaviors.StanfordProfileStylesNodeProtection = {
+    attach: function (context, settings) {
+      function addListeners() {
+        // Add listeners for all types of data entry.
+        $("input, button, :input", context).once('form-protection').on('blur change click dblclick keydown mousedown select submit', function (e) {
+          edit = true;
+          save = false;
+
+          // Node save button.
+          if ($(this).attr('value') == 'Save') {
+            save = true;
+          }
+        });
+      }
+
+      addListeners();
+      // Periodically add listeners to buttons to check for activity
+      setTimeout(addListeners, 2000);
+
+      // Handle backbutton, exit etc.
+      window.onbeforeunload = function () {
+        // Add CKEditor support.
+        if (typeof (CKEDITOR) != 'undefined' && typeof (CKEDITOR.instances) != 'undefined') {
+          for (var i in CKEDITOR.instances) {
+            if (CKEDITOR.instances[i].checkDirty()) {
+              edit = true;
+              break;
+            }
+          }
+        }
+
+        if (edit && !save) {
+          return (Drupal.t("You will lose all unsaved work."));
```

Comment:
```suggestion
          return (Drupal.t("Are you sure you want to leave this page? You will lose all unsaved work."));
```

Reply:
This doesn't actually display anyways. The first response on https://stackoverflow.com/questions/38879742/is-it-possible-to-display-a-custom-message-in-the-beforeunload-popup
>>> Chrome removed support for custom message in ver 51
Opera removed support for custom message in ver 38
Firefox removed support for custom message in ver 44.0 (still looking for source for this information)
Safari removed support for custom message in ver 9.1